File: app/src/main/java/top/easelink/lcg/LCGApp.java
package top.easelink.lcg;

import android.app.Application;
import android.content.Context;

import com.tencent.bugly.Bugly;
import com.tencent.bugly.beta.Beta;
import com.tencent.stat.StatService;

import timber.log.Timber;
import top.easelink.lcg.mta.EventHelperKt;
import uk.co.chrisjenx.calligraphy.CalligraphyConfig;

import static top.easelink.lcg.mta.MTAConstantKt.EVENT_APP_LAUNCH;

public class LCGApp extends Application {

    private static LCGApp INSTANCE;

    @Override
    public void onCreate() {
        super.onCreate();
        INSTANCE = this;
        initBulgy();
        if (BuildConfig.DEBUG) {
            Timber.plant(new Timber.DebugTree());
        }
    }

    public static Context getContext() {
        return INSTANCE;
    }
    public static LCGApp getInstance() {
        return INSTANCE;
    }

    private void initBulgy() {
        Bugly.init(getApplicationContext(), BuildConfig.BUGLY_APP_ID, false);
        Beta.largeIconId = R.drawable.ic_noavatar_middle;
        Beta.smallIconId = R.drawable.ic_noavatar_middle;
        Beta.enableHotfix = false;

        StatService.registerActivityLifecycleCallbacks(LCGApp.this);
        EventHelperKt.sendEvent(EVENT_APP_LAUNCH);
    }
}

File: app/src/main/java/top/easelink/lcg/LCGApp.kt
package top.easelink.lcg

import android.app.Application
import android.content.Context
import com.tencent.bugly.Bugly
import com.tencent.bugly.beta.Beta
import com.tencent.stat.StatService
import timber.log.Timber
import timber.log.Timber.DebugTree
import top.easelink.framework.guard.AppGuardStarter
import top.easelink.framework.log.ErrorReportTree
import top.easelink.lcg.mta.EVENT_APP_LAUNCH
import top.easelink.lcg.mta.sendEvent


class LCGApp : Application() {
    override fun onCreate() {
        super.onCreate()
        instance = this
        if (BuildConfig.DEBUG) {
            Timber.plant(DebugTree())
        } else {
            Timber.plant(ErrorReportTree())
        }

        initBulgy()
        initMTA()
        AppGuardStarter.init(this)
    }

    private fun initBulgy() {
        Bugly.init(applicationContext, BuildConfig.BUGLY_APP_ID, false)
        Beta.largeIconId = R.drawable.ic_noavatar_middle
        Beta.smallIconId = R.drawable.ic_noavatar_middle
        Beta.enableHotfix = false
    }

    private fun initMTA() {
        StatService.setContext(context)
        StatService.registerActivityLifecycleCallbacks(instance)
        // fixme incompatible with Bulgy, need more investigation
//        StatCrashReporter.getStatCrashReporter(this).apply {
//            isEnableInstantReporting = true
//            javaCrashHandlerStatus = true
//        }
        sendEvent(EVENT_APP_LAUNCH)
    }

    companion object {
        lateinit var instance: LCGApp
            private set
        @JvmStatic
        val context: Context
            get() = instance

    }
}